### ISSUE TITLE
Lightbox: caption missing when opening lightbox image

### ISSUE THREAD

**Issue by mamaretti** [WordPress/gutenberg#60469]
### Description

hello 

since #58835 figcaption is missing on opened lightbox image

(WP 6.5)

### Step-by-step reproduction instructions

add an image with caption and trigger lightbox link

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
@artemiomorales can you look into this?

**Comment by annezazu** [comment]
Pulling in feedback and a question from a separate issue:

>I am the developer of an image caption plugin for publishers who need to show image attributions for legal reasons.
With WordPress 6.5, lightboxes no longer show the caption I added to images in the post content since lightbox HTML is not rendered separately.

>Ideally, I could add a placeholder for the caption into block_core_image_print_lightbox_overlay() using a filter, but there is none.
Would you be able to provide one?

>In a next step, I would need to figure out how to fill it, but the render_block_core/image filter might just be good for that.

**Comment by artemiomorales** [comment]
@annezazu Thanks for the ping! I'm not sure I have bandwidth to look at this for the upcoming release. To clarify @webzunft, do you need to render the caption inside the lightbox overlay itself? And this was working prior to 6.5?

**Comment by webzunft** [comment]
@artemiomorales 
> do you need to render the caption inside the lightbox overlay itself? And this was working prior to 6.5?

Yes and yes. In 6.4, I just had to place my caption markup close to the image and it was showing up with it in the lightbox.

**Comment by gziolo** [comment]
It might be related to this refactoring that uses a shared HTML for all Lightbox overlays: 
- https://github.com/WordPress/gutenberg/pull/58835

**Comment by andreawetzel** [comment]
Our photographers would also like the caption to appear when the lightbox is expanded for an image.

**Comment by t-hamano** [comment]
From what I've tested, even in WP 6.4, lightbox images don't display captions because the figcaption is hidden with CSS:

https://github.com/WordPress/gutenberg/blob/wp/6.4/packages/block-library/src/image/style.scss#L276-L278

**Comment by dan-zakirov** [comment]
Opened PR #77477 which restores the caption inside the lightbox overlay and adds two Gallery-level toggles (`Show in gallery` / `Show in lightbox`) so authors can control where the caption appears.

The toggles default to on, so existing posts behave the way they did before #58835 — with the caption now actually showing up in the lightbox again. Turning **Show in gallery** off is also what unblocks the wordpress.org Plugin Directory screenshot migration (meta#8083, WordPress/wordpress.org#524), where a compact thumbnail grid with descriptions appearing only on enlarge is the target UX.

A follow-up PR is planned for per-image overrides and richer caption options (link support, positioning) — those aren't needed to close this issue and are kept out of the current PR to keep scope tight.

Reviews appreciated.

**Comment by t-hamano** [comment]
I don't mind restoring the captions, but we'll need to ensure that there are no layout issues when the captions are long. See https://github.com/WordPress/gutenberg/issues/51170

**Comment by dan-zakirov** [comment]
> I don't mind restoring the captions, but we'll need to ensure that there are no layout issues when the captions are long. See [#51170](https://github.com/WordPress/gutenberg/issues/51170)

Fair point that #51170 is a real edge case - but Id push back a bit on letting it shape this PR. Treating "long captions" as a caption problem is the wrong framing IMO: a caption is by definition a short label/description attached to an image, and multi-paragraph text is a different content type that needs a different UI (an image description field, an expandable/"Read more" pattern, or a dedicated modal/dialog separate from the lightbox overlay)

So instead of capping or restyling the caption to accommodate content that doesn't belong there, I'd suggest:

- restore the base caption behaviour here (what this PR does),
- address #51170 as a separate design task with one of the approaches above - that discussion deserves its own scope rather than gating #60469

Happy to help with the follow-up, but I don't think restricting captions is the right answer

**Comment by dknauss** [comment]
**Another concrete case for this:** [Dirtbag](https://github.com/dknauss/dirtbag) is a block theme that ships **no theme-authored JavaScript** (and no `functions.php`) by design, while enabling the core image lightbox. When a gallery image has a caption, expanding it drops the caption entirely — and with no theme JS, there's no way to restore it from the theme side. The caption is the context you most want in the enlarged view.

PR #77477 (captions in the overlay + gallery-level toggles) looks like the right fix.

**Update:** Separate from the missing-lightbox-caption bug here, I also reduced an on-page Gallery overlay-caption accessibility issue. That belongs in #56587: long overlay captions can become non-focusable scroll regions, and external/normal-flow captions are the cleaner theme-side escape hatch.

Related: #56587 and #77477
